include/mixer.h

```diff
@@ -518,30 +520,21 @@ inline void MIXER_PullFromQueueCallback(const int frames_requested, DeviceType*
 		// This provides a good size to avoid over-runs and stalls.
 		device->output_queue.Resize(iceil(device->channel->GetFramesPerBlock() * 2.0f));
 	}
-	int frames_recieved = 0;
-	if (frames_requested > 0) {
-		std::vector<AudioType> to_mix = {};
-		device->output_queue.BulkDequeue(to_mix, frames_requested);
-		frames_recieved = check_cast<int>(to_mix.size());
-		if (frames_recieved > 0) {
-			// One of the GCC CI builds throws a duplicated branch warning
-			// Clang apparently doesn't have this warning because it throws an "unknown warning" warning in the pragma
-			#ifndef __clang__
-			#pragma GCC diagnostic push
-			#pragma GCC diagnostic ignored "-Wduplicated-branches"
-			#endif
-			if (std::is_same<AudioType, AudioFrame>::value) {
-				// AudioFrame has the same memory layout as 2x floats
-				device->channel->template AddSamples<float, stereo, signeddata, nativeorder>(frames_recieved, reinterpret_cast<float*>(to_mix.data()));
-			} else {
-				device->channel->template AddSamples<AudioType, stereo, signeddata, nativeorder>(frames_recieved, to_mix.data());
-			}
-			#ifndef __clang__
-			#pragma GCC diagnostic pop
-			#endif
+	static std::vector<AudioType> to_mix = {};
+
+	const auto frames_received = check_cast<int>(
+	        device->output_queue.BulkDequeue(to_mix, frames_requested));
+
+	if (frames_received > 0) {
+		if constexpr (std::is_same_v<AudioType, AudioFrame>) {
+			device->channel->AddAudioFrames(to_mix);
+		} else {
+			device->channel->template AddSamples<AudioType, stereo, signeddata, nativeorder>(
+			        frames_received, to_mix.data());
```

Comment:
OMG, that's some C++ weirdness I haven't seen before (the "inline" `constexpr` and `template` thing). I don't even wanna know what this does... 😅 Luckily, it will all go away with the move to using floats everywhere 😆

Reply:
> it will all go away with the move to using floats everywhere

🏅 That'll be so nice to have one code path instead of all that type handling.